sklearn/compose/tests/test_column_transformer.py

```diff
@@ -571,3 +617,149 @@ def test_column_transformer_remainder_pandas(key):
+
+    ct = ColumnTransformer([('trans1', Trans(), [0])],
+                           remainder=SparseMatrixTrans())
+
+    X_trans = ct.fit_transform(X_array)
+    assert sparse.issparse(X_trans)
+    # SparseMatrixTrans creates 3 features for each column. There is
+    # one column in ``transformers``, thus:
+    assert X_trans.shape == (3, 3 + 1)
+
+    exp_array = np.hstack(
+        (X_array[:, 0].reshape(-1, 1), np.eye(3)))
+    assert_array_equal(X_trans.toarray(), exp_array)
+    assert len(ct.transformers_) == 2
+    assert ct.transformers_[-1][0] == 'remainder'
+    assert_array_equal(ct.transformers_[-1][2], [1, 2])
+
+
+def test_column_transformer_drop_all_sparse_remainder_transformer():
+    X_array = np.array([[0, 1, 2],
+                        [2, 4, 6],
+                        [8, 6, 4]]).T
+    ct = ColumnTransformer([('trans1', 'drop', [0])],
+                           remainder=SparseMatrixTrans())
+
+    X_trans = ct.fit_transform(X_array)
+    assert sparse.issparse(X_trans)
+
+    #  SparseMatrixTrans creates 3 features for each column, thus:
+    assert X_trans.shape == (3, 3)
+    assert_array_equal(X_trans.toarray(), np.eye(3))
+    assert len(ct.transformers_) == 2
+    assert ct.transformers_[-1][0] == 'remainder'
+    assert_array_equal(ct.transformers_[-1][2], [1, 2])
+
+
+def test_column_transformer_get_set_params_with_remainder():
+    ct = ColumnTransformer([('trans1', StandardScaler(), [0])],
+                           remainder=StandardScaler())
+
+    exp = {'n_jobs': 1,
+           'remainder': ct.remainder,
+           'remainder__copy': True,
+           'remainder__with_mean': True,
+           'remainder__with_std': True,
+           'trans1': ct.transformers[0][1],
+           'trans1__copy': True,
+           'trans1__with_mean': True,
+           'trans1__with_std': True,
+           'transformers': ct.transformers,
+           'transformer_weights': None}
+
+    assert ct.get_params() == exp
+
+    ct.set_params(remainder__with_std=False)
+    assert not ct.get_params()['remainder__with_std']
+
+    ct.set_params(trans1='passthrough')
+    exp = {'n_jobs': 1,
+           'remainder': ct.remainder,
+           'remainder__copy': True,
+           'remainder__with_mean': True,
+           'remainder__with_std': False,
+           'trans1': 'passthrough',
+           'transformers': ct.transformers,
+           'transformer_weights': None}
+
+    assert ct.get_params() == exp
+
+
+def test_column_transformer_remainder_transformer_error_msg_1D():
+    X_array = np.array([[0., 1., 2.], [2., 4., 6.]]).T
+
+    col_trans = ColumnTransformer([('trans', StandardScaler(), 0)],
+                                  remainder=StandardScaler())
+    assert_raise_message(ValueError, "1D data passed to a transformer",
+                         col_trans.fit, X_array)
+    assert_raise_message(ValueError, "1D data passed to a transformer",
+                         col_trans.fit_transform, X_array)
```

Comment:
I don't fully understand the reason for this test (it seems a bit superfluous because remainder is not related with the StandardScaler errorring on the 1D data)